consul.go

```diff
@@ -54,8 +54,10 @@ func (s *ConsulSource) Lock(ctx context.Context, opts *LockOptions) (<-chan bool
 		}()
 		for {
 			lockCh, err := lock.Lock(stopCh)
-			if err != nil {
-				logrus.Errorf("Error acquiring lock: %v", err)
+			if lockCh == nil {
```

Comment:
Why ar ewe now checking on the lockCh? Shouldn't we log the error and return regardless? (not sure what the API docs from consul say)

Reply:
Because lock is only acquired when Lock(stopCh) returns a non-nil `lockCh`. The diff here is to cover a case where both `lockCh` and `err` are nil upon a stop signal.

Reply:
Is there a case where lockCh != nil and err != nil though? (If so this would miss it)

Reply:
Not as of the current API, but I've just extended the condition to be independently safe.